### PR TITLE
Update ORT Mobile documentation

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Update ORT Mobile documentation for both the pre-built package and custom build usage

Add info on pre-built package and CoreML EP
Refer to operator kernels and contrib ops documentation in github so we can point to the version specific content
Tweak some aspects to improve layout like not specifying nav_order in places (items sort alphabetically by default)

Test version: https://skottmckay.github.io/onnxruntime/docs/ - in particular the contents of https://skottmckay.github.io/onnxruntime/docs/how-to/mobile/

**Motivation and Context**
Add new documentation

### CODE REVIEW

File: docs/how-to/mobile/overview.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The pre-built package includes support for selected operators and ONNX opset versions based on the requirements for popular models. If you choose to use the pre-built package you will not need a development environment to perform a custom build of ONNX Runtime, however the binary size will be larger than if you do a custom build with just the operators required by your model/s. Your model can only use the opsets and operators supported by the pre-built package. 
+
+
+### Available pre-built packages
+
+| Platform | Package location | Included Execution Providers |
+|----------|------------------|----------|
+| Android | onnxruntime-mobile package in Maven  | CPU Execution Provider <br/>NNAPI Execution Provider |
+| iOS (preview) | onnxruntime-mobile CocoaPod (C/C++ APIs) <br>onnxruntime-mobile-objc CocoaPod (Objective-C API) | CPU Execution Provider <br/>CoreML Execution Provider |
```
Comment (gwang-msft) [inline docs/how-to/mobile/overview.md:34]: Are we include CoreML in the pre-built package?

File: docs/how-to/mobile/model-execution.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+If you wish to explicitly say that the InferenceSession input is an ORT format model you can do so via SessionOptions, although this generally should not be necessary.
+
+C++ API
+```C++
+Ort::SessionOptions session_options;
+session_options.AddConfigEntry('session.load_model_format', 'ORT');
+```
+
+Python
```
Comment (gwang-msft) [inline docs/how-to/mobile/model-execution.md:38]: Remove python and add java?

File: docs/how-to/mobile/overview.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+parent: Deploy ONNX Runtime Mobile
+grand_parent: How to
+has_children: false
+nav_order: 1
+---
+{::options toc_levels="2" /}
+
+# Overview
+
+ONNX Runtime Mobile is a special build of ONNX Runtime that minimize the binary size and reduces memory usage. It does this primarily by only including specified operators and types in the build, and by saving a pre-optimized ONNX model to an internal format ('ORT format model').
```
Comment (edgchen1) [inline docs/how-to/mobile/overview.md:12]: >minimize [](http://example.com/codeflow?start=60&length=8)

"minimize" -> "minimizes"

File: docs/reference/execution-providers/CoreML-ExecutionProvider.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+uint32_t coreml_flags = 0;
+Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CoreML(so, coreml_flags));
+Ort::Session session(env, model_path, so);
+```
+
+## Configuration Options
+
+There are several run time options available for the CoreML EP.
+
+To use the CoreML EP run time options, create an unsigned integer representing the options, and set each individual options by using the bitwise OR operator.
```
Comment (edgchen1) [inline docs/reference/execution-providers/CoreML-ExecutionProvider.md:48]: >options [](http://example.com/codeflow?start=116&length=7)

"options" -> "option"

Comment (edgchen1) [reply docs/reference/execution-providers/CoreML-ExecutionProvider.md:48]: also in NNAPI EP doc

File: Gemfile.lock [hunk 5]
```diff
@@ -25,10 +25,12 @@ GEM
     ethon (0.12.0)
       ffi (>= 1.3.0)
     eventmachine (1.2.7)
+    eventmachine (1.2.7-x64-mingw32)
```
Comment (edgchen1) [inline Gemfile.lock:28]: should these changes be checked in?

File: docs/resources/mobile-performance-tuning.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+The _extended_ level optimizations introduce the custom FusedConv nodes, which the NNAPI EP ignores as it will only take nodes that are using ONNX operators that NNAPI can handle. This results in two nodes using NNAPI, each handling a single MaxPool operation. The performance of this model is likely to be adversely affected, as the overhead of the device copies between CPU and NNAPI (which are required before and after each of the two NNAPI nodes) is unlikely to be exceeded by the time saved executing a single MaxPool operation each time using NNAPI. Better performance may be obtainable by not registering the NNAPI EP so that all nodes in the model are executed using the CPU EP.
+
+![Changes to nodes by NNAPI EP depending on the optimization level the model was created with](../../images/mnist_optimization_with_nnapi.png)
+
+## 2. Initial Performance Testing
+
+The best optimization settings will differ by model. Some models may perform better with NNAPI, some models may not. As the performance will be model specific you must performance test to determine the best combination for your model.
```
Comment (YUNQIUGUO) [inline docs/resources/mobile-performance-tuning.md:75]: "you must performance test" -> "you must run performance tests"?

**Review by edgchen1 — APPROVED** [review]

File: Gemfile.lock (hunk 5 above)
Comment (skottmckay) [reply Gemfile.lock:28]: They're automatically added when run on a windows machine. They don't seem to cause an issue when generating the pages on github so I left them in there.

File: docs/how-to/mobile/initial-setup.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    ```
+    repositories {
+        mavenCentral()
+    }
+    ```
+
+2. build.gradle (Module):
+    ```
+    dependencies {
+        compile 'com.microsoft.onnxruntime:onnxruntime-mobile:1.8.0'
```
Comment (gwang-msft) [inline docs/how-to/mobile/initial-setup.md:35]: My mistake, this should be 
```
compile 'com.microsoft.onnxruntime:onnxruntime-mobile:1.8.0'
```

**Pull request merged**